### PR TITLE
Feat/sign up and in

### PR DESCRIPTION
**Pull request by hhmehsan**
i use style tag and normal css for .router-link-exact-active is that ok? or tailwind has something for that?

### CODE REVIEW

File: src/views/SignIn.vue [hunk 1]
```diff
@@ -4,31 +4,37 @@
...
       
       <div class="w-9/12 sm:w-7/12 lg:w-5/12 mx-auto mt-20">
-        <p class="text-5xl text-center mb-9 text-brwn font-semibold">Welcome Back! </p>
-        <form class="mt-20">
+        <p class="text-4xl md:text-5xl text-center mb-9 text-brwn font-semibold">Welcome Back! </p>
+        <Form class="mt-20" :validation-schema="schema">
             
          
           
           <div class="mb-6">
```
Comment (amE80) [inline src/views/SignIn.vue:12]: You use mb-6 for all of your inputs section to have some space between them . that's cool and there is no problem with it , but if you have a lot of inputs or lists , you need to put mb-6 classes to all of them , but in tailwind there is a class called space-x- or space-y- which you can use to have spaces between child elements , it's easy to use , you put this class to the parent element and spaces will be applied to all of child elements . 
read it from : https://tailwindcss.com/docs/space

File: src/includes/validation.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            generateMessage: (ctx)=>{
+                const messages ={
+                    required: ` You must fill ${ctx.field} field! `,
+                    email:`it isn't a correct email address`,
+                    max: `it's too long`,
+                    min: `it's too short`,
+                };
+                const message = messages[ctx.rule.name] ? messages[ctx.rule.name]: `this field is invalid`;
+                return message;
+            }
```
Comment (amE80) [inline src/includes/validation.js:25]: validateOnBlur:true,
 validateOnChange: true,
 validateOnInput : true,
 validateOnModelUpdate: true
 
 Put these codes here to active the behavior which you want to have for your validation (real time validation)

File: src/components/Navigation/topNav.vue [hunk 3]
```diff
@@ -3,9 +3,9 @@
         <ul class="flex items-center justify-between">
              <li class="text-2xl font-bold md:text-3xl text-brwn "> <h2> Blog Hub </h2></li>
            <ul class="flex items-center justify-between w-40 sm:w-48">      
-            <li @click="goToHome" class="text-sm cursor-pointer md:text-base hover:text-brwn font-main text-lightGray">Home </li>
-            <li @click="goToSignIn" class="text-sm cursor-pointer md:text-base hover:text-brwn font-main text-lightGray" >Sign in </li>
-            <li @click="goToSignUp" class="text-sm cursor-pointer md:text-base hover:text-brwn font-main text-lightGray">Sign up </li>
+            <li  class="text-sm cursor-pointer md:text-base hover:text-brwn font-main text-lightGray"><router-link class=" " :to="{name: 'home'}"> Home </router-link></li>
```
Comment (amE80) [inline src/components/Navigation/topNav.vue:6]: you can add for example ==>  active-class="text-blue-600 bg-slate-50" into your router link to have the active style for it : pay attention , active-class is reserved word you need to write it as same as here , but bg-slate-50 and other styles are optional , you can have any styles which you want.

**Review by amE80 — APPROVED** [review]

**Comment by amE80** [comment]
Everything looks great , I left  you some comments  but this merge request won't be drafted , because there was no problem in your codes . I commented how you can use active style and real time validation.
Good luck and good job.

**Comment by hhmehsan** [comment]
thank you darling

**Pull request merged**